### PR TITLE
[i18n/Audio] Adding renderers and helpers for remaining election strings

### PR DESCRIPTION
**Pull request by kofi-q**
## Overview

_Easier by commit_

_Related spec [here](https://docs.google.com/document/d/1a2GRj1g4aOawEssPdTCgbv-qs28d-M_dyMEfzm4I4nM/edit?usp=sharing)._

Fleshing out `@votingworks/ui/electionStrings` to include all the translatable/spoken strings in the election definition so far.

Making the `VxMark` `StartPage` language-aware to test out the new election strings, which led to adding support for number and date i18n -- the audio piece of that is going to require a bit more planning and coordination with the VxDesign work, so I'll start writing up some thoughts in the spec.
- Also, to avoid audio interpolation, the previous `"Your ballot contains x contests."` string is now `"Number of contests on your ballot: x"`

## Demo Video or Screenshot

https://github.com/votingworks/vxsuite/assets/264902/481d65a7-4c80-4d98-bcdd-3794d9613fca

## Testing Plan
- Unit tests for the new ui string utils
- Updated expectations in existing tests for changed strings/HTML structures
- Manually tested `StartPage` in different languages with mock translations.

## Checklist

- ~[ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.~

- [x] I have added a screenshot and/or video to this PR to demo the change
- ~[x] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates~

### CODE REVIEW

File: libs/ui/src/ui_strings/number_string.test.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    <H1>
+      <NumberString value={100000} />
+    </H1>
+  );
+
+  await screen.findByRole('heading', { name: '100,000' });
+
+  act(() => getLanguageContext()?.setLanguage(LanguageCode.SPANISH));
+
+  await screen.findByRole('heading', { name: '100.000' });
```
Comment (arsalansufi) [inline libs/ui/src/ui_strings/number_string.test.tsx:27]: I've always thought the smaller `.` for aesthetic separator and bigger `,` for decimal separator made more sense lol

File: libs/ui/src/ui_strings/election_strings.tsx [hunk 2]
```diff
@@ -1,22 +1,104 @@
-import { Party } from '@votingworks/types';
+/* eslint-disable react/destructuring-assignment */
```
Comment (arsalansufi) [inline libs/ui/src/ui_strings/election_strings.tsx:1]: Just curious, why is this necessary?

File: libs/ui/src/ui_strings/utils.tsx [hunk 3]
```diff
@@ -29,3 +40,32 @@ export function renderCandidatePartyList(
     </React.Fragment>
   );
 }
+
+export function renderPrecinctSelectionName(
+  electionPrecincts: readonly Precinct[],
+  precinctSelection?: PrecinctSelection
+): React.ReactNode {
```
Comment (arsalansufi) [inline libs/ui/src/ui_strings/utils.tsx:47]: Any particular reason for making these helpers functions that return JSX instead of components? E.g. this could alternatively be expressed as:

```jsx
<PrecinctSelectionName
  electionPrecincts={electionPrecincts}
  precinctSelection={precinctSelection}
/>
```

**Review by arsalansufi — APPROVED** [review]
LGTM!

File: libs/ui/src/ui_strings/election_strings.tsx (hunk 2 above)
Comment (kofi-q) [reply libs/ui/src/ui_strings/election_strings.tsx:1]: Not sure why this happened, but once I switched over to using the `ElectionStringKey` enum as the keys, the eslint rule started interpreting the method params as `props` for components, so it complained about not destructuring `candidate` before referencing `candidate.id`, for example.

File: libs/ui/src/ui_strings/number_string.test.tsx (hunk 1 above)
Comment (kofi-q) [reply libs/ui/src/ui_strings/number_string.test.tsx:27]: same!

File: libs/ui/src/ui_strings/utils.tsx (hunk 3 above)
Comment (kofi-q) [reply libs/ui/src/ui_strings/utils.tsx:47]: oh, yeah! thanks for flagging - i initially had `candidatePartyList` as a method in the `electionStrings` mapping and didn't rethink the pattern when i broke it out into a separate function. these make more sense as components

File: libs/ui/src/ui_strings/election_strings.tsx (hunk 2 above)
Comment (arsalansufi) [reply libs/ui/src/ui_strings/election_strings.tsx:1]: Got it, thanks for explaining

File: libs/ui/src/ui_strings/utils.tsx (hunk 3 above)
Comment (arsalansufi) [reply libs/ui/src/ui_strings/utils.tsx]: Makes sense!
